src/utils/transaction/userOperation.ts

```diff
@@ -1,134 +0,0 @@
-import {
-  getAddress,
-  getContract,
-  prepareContractCall,
-  readContract,
-  type Address,
-  type Hex,
-} from "thirdweb";
-import {
-  createUnsignedUserOp,
-  getPaymasterAndData,
-  signUserOp,
-  type UserOperation,
-} from "thirdweb/wallets/smart";
-import { getAccount } from "../account";
-import { getChain } from "../chain";
-import { thirdwebClient } from "../sdk";
-import type { PopulatedTransaction, QueuedTransaction } from "./types";
-
-export const generateSignedUserOperation = async (
-  queuedTransaction: QueuedTransaction,
-  populatedTransaction: PopulatedTransaction,
-): Promise<UserOperation> => {
-  const {
-    chainId,
-    signerAddress,
-    accountAddress,
-    accountFactoryAddress: userProvidedAccountFactoryAddress,
-    target,
-    from,
-    overrides,
-  } = queuedTransaction;
-
-  if (!from || !accountAddress || !signerAddress || !target) {
-    throw new Error("Invalid UserOperation parameters");
-  }
-
-  const chain = await getChain(chainId);
-
-  // Resolve Smart-Account Contract
-  const smartAccountContract = getContract({
-    client: thirdwebClient,
-    chain,
-    address: accountAddress as Address,
-  });
-
-  // use the user provided factory address if available
-  let accountFactoryAddress = userProvidedAccountFactoryAddress;
-
-  if (!accountFactoryAddress) {
-    // Resolve Factory Contract Address from Smart-Account Contract
-    try {
-      const onchainAccountFactoryAddress = await readContract({
-        contract: smartAccountContract,
-        method: "function factory() view returns (address)",
-        params: [],
-      });
-
-      accountFactoryAddress = getAddress(onchainAccountFactoryAddress);
-    } catch {
-      // if no factory address is found, throw an error
-      throw new Error(
-        `Failed to find factory address for account '${accountAddress}' on chain '${chainId}'`,
-      );
-    }
-  }
-
-  // Resolve Factory Contract
-  const accountFactoryContract = getContract({
-    client: thirdwebClient,
-    chain,
-    address: accountFactoryAddress as Address,
-  });
-
-  // Prepare UserOperation Call
-  const userOpCall = prepareContractCall({
-    contract: smartAccountContract,
-    method: "function execute(address, uint256, bytes)",
-    params: [
-      populatedTransaction.to || "",
-      populatedTransaction.value || 0n,
-      populatedTransaction.data,
-    ],
-  });
-
-  // Create Unsigned UserOperation
-  // Todo: Future expose a way to skip paymaster
-  const unsignedOp = (await createUnsignedUserOp({
-    transaction: userOpCall,
-    accountContract: smartAccountContract,
-    sponsorGas: true,
-    factoryContract: accountFactoryContract,
-    adminAddress: signerAddress,
-  })) as UserOperation; // TODO support entrypoint v0.7 accounts
-
-  // Handle gas limit override.
-  if (overrides?.gas) {
```

Comment:
As per @0xFirekeeper, this is being used by one of our large customers. We may still need gas override options in userOps.

Reply:
Handled in the SDK now.

It will respect whatever gas you set in the initial tx you're trying to bundle